Sources/AWSLambdaRuntimeCore/LambdaHandler.swift

```diff
@@ -129,18 +129,42 @@ public protocol EventLoopLambdaHandler: ByteBufferLambdaHandler {
 public extension EventLoopLambdaHandler {
     /// Driver for `ByteBuffer` -> `In` decoding and `Out` -> `ByteBuffer` encoding
     func handle(context: Lambda.Context, event: ByteBuffer) -> EventLoopFuture<ByteBuffer?> {
-        switch self.decodeIn(buffer: event) {
+        let segment = context.tracer.beginSegment(name: "HandleEvent", baggage: context.baggage)
+        // TODO: record errors propagated in result types?
+        let decodedEvent = segment.subsegment(name: "DecodeIn") { _ in
```

Comment:
btw, I've been wondering if we should offer `tracer.withSpan() { ... }` as built-in for wrapping synchronous blocks with a span in the swift-tracing API right away, or if we should stay away from adding any kind of sugar in the API package.

This is quite common I think so I think we could add it...

We could also do the same with a NIO extensions package then to handle Future returning blocks 🤔 

WDYT @pokryfka @tomerd ?

Reply:
@ktoso +1

Reply:
I do use "helpers" both with closures and NIO futures as referenced here https://github.com/slashmo/gsoc-swift-tracing/issues/125#issuecomment-674356002 
when API is changed to `TracingInstrument` the amount of sugar will depend on what `swift-tracing` provides